cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunnerTest.java

```diff
@@ -350,12 +361,18 @@ public void apply() {
 
   @Test
   public void testWordCount() throws Exception {
-
+    // deploy to namespace default by default
     final ApplicationWithPrograms app = deployApp(AppWithMapReduce.class);
     final String inputPath = createInput();
     final java.io.File outputDir = new java.io.File(TEMP_FOLDER.newFolder(), "output");
 
-    final KeyValueTable jobConfigTable = datasetCache.getDataset("jobConfig");
+    try {
+      final KeyValueTable jobConfigTable = datasetCache.getDataset("someOtherNameSpace", "jobConfig");
+    } catch (DatasetInstantiationException e) {
+      Assert.assertTrue(e.getMessage().equals("Could not instantiate dataset 'someOtherNameSpace:jobConfig'"));
+    }
+    // Should work if explicitly specify the default namespace
+    final KeyValueTable jobConfigTable = datasetCache.getDataset("default", "jobConfig");
```

Comment:
Strictly speaking, this should be NamespaceId.DEFAULT.getNamespace() instead of hardcoding "default"